### PR TITLE
Added find_if and find_if_not tests

### LINKED ISSUE #194: Add find_if tests
From #193 by @Guekka:

> Sidenote: while writing this, I've noticed find_if doesn't seem tested?

Embarrassingly, it's true, we don't have specific tests for `find_if` and `find_if_not` (although they're used in the implementation of `filter` and various other places, so we know they work...)

The lack of tests should be fixed.

### PR DESCRIPTION
**Pull request by nwad123**
Resolves #194

### CODE REVIEW

**Comment by nwad123** [comment]
Hi! This is my first time attempting to contribute to a project on GitHub so if I'm doing anything wrong please let me know! I followed some instructions I found online for submitting a pull request, and I'm not actually confident that I did it right, so if there is a better process I'd love to get a quick walkthrough.

File: test/test_find_if_not.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+// Incompatible predicate type
+using S_comp = auto(S x) -> bool;
+static_assert(not std::invocable<find_if_not_fn, int[10], S_comp>);
+
+// Not a predicate
+using not_a_predicate = auto(int x) -> S;
+static_assert(not std::invocable<find_if_not_fn, int[10], not_a_predicate>);
+
+constexpr bool test_find_if()
```
Comment (tcbrindle) [inline test/test_find_if_not.cpp:47]: This is testing `find_if_not()` :)

```suggestion
constexpr bool test_find_if_not()
```

File: test/test_find_if_not.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        cur = lens.find_if_not(is_0_1_2_or_3);
+        if (cur != 4) {
+            return false;
+        }
+    }
+
+    return true;
+}
+static_assert(test_find_if());
```
Comment (tcbrindle) [inline test/test_find_if_not.cpp:102]:
```suggestion
static_assert(test_find_if_not());
```

File: test/test_find_if.cpp [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+} // namespace
+
+TEST_CASE("find_if")
+{
+    {
+        std::vector<int> vec {1, 2, 3, 4, 5};
+        auto is_greater_than_3 = [](int x) { return x > 3; };
+        auto idx = flux::find_if(vec, is_greater_than_3);
+        REQUIRE(idx == 3);
+    }
```
Comment (tcbrindle) [inline test/test_find_if.cpp:113]: We want to run the static tests at runtime as well, so we need to include them in the `TEST_CASE` block

```suggestion
    REQUIRE(test_find_if());

    {
        std::vector<int> vec {1, 2, 3, 4, 5};
        auto is_greater_than_3 = [](int x) { return x > 3; };
        auto idx = flux::find_if(vec, is_greater_than_3);
        REQUIRE(idx == 3);
    }
```

File: test/test_find_if_not.cpp [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+} // namespace
+
+TEST_CASE("find_if_not")
+{
+    {
+        std::vector<int> vec {1, 2, 3, 4, 5};
+        auto is_odd = [](int x) { return x % 2 == 1; };
+        auto idx = flux::find_if_not(vec, is_odd);
+        REQUIRE(idx == 1);
+    }
```
Comment (tcbrindle) [inline test/test_find_if_not.cpp:113]: As above, we want to run the static test in the `TEST_CASE` block here as well

```suggestion

    REQUIRE(test_find_if_not());

    {
        std::vector<int> vec {1, 2, 3, 4, 5};
        auto is_odd = [](int x) { return x % 2 == 1; };
        auto idx = flux::find_if_not(vec, is_odd);
        REQUIRE(idx == 1);
    }
```

**Review by tcbrindle — CHANGES_REQUESTED** [review]
This is very comprehensive, thank you! Just a couple of little comments -- the function statically testing `find_if_not()` should be called `test_find_if_not()`, and we want to run the static test functions at runtime as well.

Other than that, this looks great!

**Review by tcbrindle — APPROVED** [review]

**Comment by tcbrindle** [comment]
Merged, thank you very much @nwad123!

**Pull request merged**